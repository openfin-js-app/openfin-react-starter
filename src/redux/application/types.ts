export interface ApplicationNewSnackbarOption {
    message:string;
    variant?:'primary'|'success'|'warning'|'error'|'info';
}

export interface ApplicationSetSnackbarStatusOption {
    open:boolean;
}

export interface ApplicationCloseSnackbarOption {
    event?:any;
    reason:string;
}

export interface SnackBarMsg {
    key:number;
    message:string;
    variant:'primary'|'success'|'warning'|'error'|'info';
}

export interface ApplicationState {
    username:string;
    computerName:string;
    deviceId:string;
    deviceUserId:string;
    loading:boolean;
    drawerOpen:boolean;
    launchBarCollapse:boolean;
    snackBarOpen:boolean;
    snackBarMsgInfo:Partial<SnackBarMsg>;
    snackBarMsgQueue:Partial<SnackBarMsg>[];
    openfinVersion:string;
    openfinHostSpec:any;
    windowsState:string;
}