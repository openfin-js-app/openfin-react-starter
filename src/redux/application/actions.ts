import { createAction, ActionFunctionAny, Action } from 'redux-actions';
import { IApplicationNewSnackbarOption, IApplicationSetSnackbarStatusOption, IApplicationCloseSnackBarOption } from './types';

export const APPLICATION_STARTED:string = 'APPLICATION_STARTED';
export const APPLICATION_READY:string = 'APPLICATION_READY';
export const APPLICATION_DRAWER_TOGGLE:string = 'APPLICATION_DRAWER_TOGGLE';
export const APPLICATION_TOGGLE_WINDOW_STATE:string = 'APPLICATION_TOGGLE_WINDOW_STATE';

export const applicationStarted:ActionFunctionAny<Action<void>> = createAction(APPLICATION_STARTED);
export const applicationReady:ActionFunctionAny<Action<void>> = createAction(APPLICATION_READY);
export const applicationDrawerToggle:ActionFunctionAny<Action<void>> = createAction(APPLICATION_DRAWER_TOGGLE);
export const applicationToogleWindowState:ActionFunctionAny<Action<void>> = createAction(APPLICATION_TOGGLE_WINDOW_STATE);

// snackbar

export const APPLICATION_NEW_SNACKBAR='APPLICATION_NEW_SNACKBAR';
export const APPLICATION_SET_SNACKBAR_STATUS='APPLICATION_SET_SNACKBAR_STATUS';
export const APPLICATION_PROCESS_SNACKBAR_QUEUE='APPLICATION_PROCESS_SNACKBAR_QUEUE';
export const APPLICATION_CLOSE_SNACKBAR='APPLICATION_CLOSE_SNACKBAR';

export const applicationNewSnackbar:ActionFunctionAny<Action<IApplicationNewSnackbarOption>>
    = createAction(APPLICATION_NEW_SNACKBAR, (option:IApplicationNewSnackbarOption)=>(option));
export const applicationSetSnackbarStatus:ActionFunctionAny<Action<IApplicationSetSnackbarStatusOption>>
    = createAction(APPLICATION_SET_SNACKBAR_STATUS, (option:IApplicationSetSnackbarStatusOption)=>(option));
export const applicationProcessSnackbarQueue:ActionFunctionAny<Action<void>>
    = createAction(APPLICATION_PROCESS_SNACKBAR_QUEUE);
export const applicationCloseSnackbar:ActionFunctionAny<Action<IApplicationCloseSnackBarOption>>
    = createAction(APPLICATION_CLOSE_SNACKBAR, (option)=>(option));


// launch bar

export const APPLICATION_LAUNCH_BAR_TOGGLE = 'APPLICATION_LAUNCH_BAR_TOGGLE';
export const APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE = 'APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE';
export const APPLICATION_LAUNCH_NEW_WINDOW = 'APPLICATION_LAUNCH_NEW_WINDOW';
export const applicationLaunchBarToggle:ActionFunctionAny<Action<void>> = createAction(APPLICATION_LAUNCH_BAR_TOGGLE);
export const applicationLaunchBarToggleCollapse:ActionFunctionAny<Action<void>> = createAction(APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE);
export const applicationLaunchNewWindow:ActionFunctionAny<Action<any>>
    = createAction(APPLICATION_LAUNCH_NEW_WINDOW,(appJson:any)=>(appJson));