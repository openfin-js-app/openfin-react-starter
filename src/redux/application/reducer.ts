import { handleActions, Action } from 'redux-actions';
import { System, Window } from '@albertli/redux-openfin';

import {
    ApplicationNewSnackbarOption,
    ApplicationSetSnackbarStatusOption,
    SnackBarMsg, ApplicationState,
} from './types';

import {
    APPLICATION_READY,
    APPLICATION_DRAWER_TOGGLE,
    APPLICATION_NEW_SNACKBAR,
    APPLICATION_SET_SNACKBAR_STATUS,
    APPLICATION_PROCESS_SNACKBAR_QUEUE,
    APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE,
} from './actions';

const defaultState:Partial<ApplicationState>={
    username:'',
    computerName:'',
    deviceId:null,
    deviceUserId:null,
    loading:true,
    drawerOpen:true,
    launchBarCollapse:false,
    snackBarOpen:false,
    snackBarMsgInfo:{},
    snackBarMsgQueue:[],
    openfinVersion:'n/a',
    openfinHostSpec:{},
    windowsState:'normal',
};

export default handleActions({
    [System.actions.GET_DEVICE_ID_RES]:(state,action)=>{
        const {id} = action.payload as any;
        return {
            ...state,
            deviceId:id
        };
    },
    [System.actions.GET_DEVICE_USER_ID_RES]:(state,action)=>{
        const {id} = action.payload as any;
        return {
            ...state,
            deviceUserId:id
        };
    },
    [System.actions.GET_ENVIRONMENT_VARIABLE_RES]:(state,action)=>{
        const {env,value} = action.payload as any;
        if (env==='username'){
            return {
                ...state,
                username:value,
            }
        }else if (env === 'computername'){
            return {
                ...state,
                computerName:value,
            }
        }else{
            return state;
        }
    },
    [System.actions.GET_VERSION_RES]:(state,action)=>{
        const {version} = action.payload as any;
        return {
            ...state,
            openfinVersion:version,
        };

    },
    [System.actions.GET_HOST_SPECS_RES]:(state,action)=>{
        const openfinHostSpec = action.payload as any;
        return {
            ...state,
            openfinHostSpec
        };

    },
    [Window.actions.GET_STATE_RES]:(state,action)=>{
        const payload = action.payload as any;
        return {
            ...state,
            windowState:payload.state,
        };

    },
    [APPLICATION_READY]:(state,action)=>({
        ...state,
        loading:false,
    }),
    [APPLICATION_DRAWER_TOGGLE]:(state,action)=>({
        ...state,
        drawOpen:!state.drawerOpen
    }),
    [APPLICATION_NEW_SNACKBAR]:(state,action)=>{
        let option:ApplicationNewSnackbarOption = action.payload as ApplicationNewSnackbarOption;
        let newMsgQueue = state.snackBarMsgQueue.concat([{
            message:option.message,
            key: new Date().getTime(),
            variant:option.variant,
        }]);
        return{
            ...state,
            snackBarMsgQueue:newMsgQueue,
        }
    },
    [APPLICATION_PROCESS_SNACKBAR_QUEUE]:(state,action)=>{
        if (state.snackBarMsgQueue.length > 0){
            let newMsg = state.snackBarMsgQueue[0];
            let newMsgQueue = state.snackBarMsgQueue.slice(1);
            return {
                ...state,
                snackBarOpen:true,
                snackBarMsgInfo: newMsg,
                snackBarMsgQueue: newMsgQueue,
            }
        }else{
            return state;
        }
    },
    [APPLICATION_SET_SNACKBAR_STATUS]:(state,action)=>{
        let option:ApplicationSetSnackbarStatusOption = action.payload as ApplicationSetSnackbarStatusOption;
        return {
            ...state,
            snackBarOpen:option.open,
        };
    },
    [APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE]:(state,action)=>({
        ...state,
        launchBarCollapse:!state.launchBarCollapse,
    })
},defaultState);