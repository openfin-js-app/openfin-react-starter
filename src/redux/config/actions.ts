import {createAction, ActionFunctionAny, Action} from 'redux-actions';

export const CONFIG_RESET:string = 'CONFIG_RESET';
export const CONFIG_UPDATE_ONE_FIELD:string = 'CONFIG_UPDATE_ONE_FIELD';

export const configReset = createAction(CONFIG_RESET, (option)=>(option));
export const configUpdateOneField = createAction(CONFIG_UPDATE_ONE_FIELD, (option)=>(option));

// optional actions depending on the config application values

export const CONFIG_UPDATE_NEW_WINDOW_POSITION:string = 'CONFIG_UPDATE_NEW_WINDOW_POSITION';
export const configUpdateNewWindowPosition = createAction(CONFIG_UPDATE_NEW_WINDOW_POSITION);