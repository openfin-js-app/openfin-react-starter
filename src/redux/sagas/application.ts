import { buffers, delay } from 'redux-saga';
import { all, call, put, take, takeLatest, takeEvery, fork, select, actionChannel } from 'redux-saga/effects';
import { System, Event, Window } from '@albertli90/redux-openfin';

import hist from '../../utils/history';

import { launchBarItems } from '../../layouts/LaunchBar/LaunchBarData';

const ENABLE_LOADING_VIEW=process.env.REACT_APP_ENABLE_LOADING_VIEW.toLowerCase() === 'true';

const LOADING_BANNER_WIDTH = parseInt(process.env.REACT_APP_LOADING_BANNER_WIDTH, 10);
const LOADING_BANNER_HEIGHT = parseInt(process.env.REACT_APP_LOADING_BANNER_HEIGHT, 10);
const DEFAULT_WIDTH = parseInt(process.env.REACT_APP_DEFAULT_APP_WIDTH, 10);
const DEFAULT_HEIGHT = parseInt(process.env.REACT_APP_DEFAULT_APP_HEIGHT, 10);

const previousBaseWindow={
    url:null,
    top:null,
    left:null,
    width:null,
    height:null,
};

import {
    APPLICATION_STARTED,
    APPLICATION_NEW_SNACKBAR,
    APPLICATION_CLOSE_SNACKBAR,
    APPLICATION_TOGGLE_WINDOW_STATE,
    applicationReady,
    applicationSetSnackbarStatus,
    applicationProcessSnackbarQueue,
    APPLICATION_LAUNCH_BAR_TOGGLE,
    APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE,
    APPLICATION_LAUNCH_NEW_WINDOW,
} from '..';

import { configUpdateNewWindowPosition } from '..';

export const getLaunchBarCollapse = state => state.application.launchBarCollapse;
export const getWindowState = state => state.application.windowState;
export const getNewWindowTop = state => state.config.application.newWinTop;
export const getNewWindowLeft = state => state.config.application.newWinLeft;
export const getNewWindowWidth = state => state.config.application.newWinWidth;
export const getNewWindowHeight = state => state.config.application.newWinHeight;

export function* handleRedirectToLoadingView(monitorRect) {
    yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
        options:{resizable:false}
    }));

    yield call(Window.asyncs.setBounds,Window.actions.setBounds({
        left:(monitorRect.right - monitorRect.left)/2 - LOADING_BANNER_WIDTH/2,
        top:(monitorRect.bottom - monitorRect.top)/2 - LOADING_BANNER_HEIGHT/2,
        width:LOADING_BANNER_WIDTH,
        height: LOADING_BANNER_HEIGHT,
    }));
}

export function* handleRedirectFromLoadingView(monitorRect) {
    // after the sagas loaded, redirect to default page/view
    if (process.env.REACT_APP_DEFAULT_VIEW_URL && process.env.REACT_APP_DEFAULT_VIEW_URL.length > 0){
        if (process.env.NODE_ENV !== 'test'){
            hist.push(process.env.REACT_APP_DEFAULT_VIEW_URL);
        }
        yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
            options:{
                resizable:true,
            }
        }));

        yield call(Window.asyncs.setBounds,Window.actions.setBounds({
            left:(monitorRect.right - monitorRect.left)/2 - DEFAULT_WIDTH/2,
            top:(monitorRect.bottom - monitorRect.top)/2 - DEFAULT_HEIGHT/2,
            width:DEFAULT_WIDTH,
            height: DEFAULT_HEIGHT,
        }));

    }else{
        // switch to launchBar
        const launchBarCollapse = yield select(getLaunchBarCollapse);

        yield call(Window.asyncs.updateOptions,Window.actions.updateOptions({
            options:{resizable:false}
        }));

        previousBaseWindow.url='/dashboard/view-one';
        previousBaseWindow.top=(monitorRect.bottom - monitorRect.top)/2 - DEFAULT_HEIGHT/2;
        previousBaseWindow.left=(monitorRect.right - monitorRect.left)/2 - DEFAULT_WIDTH/2;
        previousBaseWindow.width=DEFAULT_WIDTH;
        previousBaseWindow.height=DEFAULT_HEIGHT;

        if(launchBarCollapse){
            yield call(Window.asyncs.setBounds,Window.actions.setBounds({
                left:(monitorRect.right - monitorRect.left)/2,
                top:(monitorRect.bottom - monitorRect.top)/4,
                width:88,
                height:64,
            }));
        }else{
            yield call(Window.asyncs.setBounds,Window.actions.setBounds({
                left:(monitorRect.right - monitorRect.left)/2,
                top:(monitorRect.bottom - monitorRect.top)/4,
                width:launchBarItems.length<10?launchBarItems.length*64+88:664,
                height:64,
            }));
        }
        if (process.env.NODE_ENV !== 'test'){
            hist.push('/launchBar');
        }

    }
}

export function* handleApplicationLoading() {

    const currentIsLoadingView =
        (new URL(window.location.href).pathname.indexOf('loading')>-1) ||
        (new URL(window.location.href).pathname.indexOf('index.html')>-1);


    const monitorInfoAction = yield call(System.asyncs.getMonitorInfo,System.actions.getMonitorInfo({}));
    const monitorRect = monitorInfoAction.payload.primaryMonitor.monitorRect;

    yield call(Window.asyncs.setAsForeground,Window.actions.setAsForeground({}));

    if (ENABLE_LOADING_VIEW && currentIsLoadingView){
        yield* handleRedirectToLoadingView(monitorRect) as any;
    }

    yield all([
        call(System.asyncs.getMachineId,System.actions.getMachineId({})),
        // getDeviceUserId might fail, thus use flux syntax........
        put.resolve(System.actions.getDeviceUserId({})),
        take(System.actions.GET_DEVICE_USER_ID_RES),
        call(System.asyncs.getEnvironmentVariable,System.actions.getEnvironmentVariable({env:'USERNAME'})),
        call(System.asyncs.getEnvironmentVariable,System.actions.getEnvironmentVariable({env:'computername'})),
        call(System.asyncs.getEnvironmentVariable,System.actions.getEnvironmentVariable({env:'HOSTNAME'})),
        call(System.asyncs.getVersion,System.actions.getVersion({})),
        call(System.asyncs.getHostSpecs,System.actions.getHostSpecs({})),
        call(Window.asyncs.getState,Window.actions.getState({})),
        // delay for loading view render, could be removed
        call(delay,5000),
    ]);

    yield put.resolve(applicationReady());

    if (ENABLE_LOADING_VIEW && currentIsLoadingView){
        yield* handleRedirectFromLoadingView(monitorRect) as any;
    }
}

export function* handleApplicationExit() {
    // -------------------------------start of app codes -----------------------------------------------

    // do something cleaning up before shutdonw~

    // ---------------------------------end of app codes -----------------------------------------------


    yield put.resolve(Window.actions.close({force:true}));
}

export function* handleApplicationAddNewSnackBar() {
    const state = yield select();
    if(state.snackBarOpen){
        yield put(applicationSetSnackbarStatus({open:false}));
    }else{
        yield put(applicationProcessSnackbarQueue());
    }
}

export function* handleApplicationCloseSnackBar(action) {
    const {event, reason} = action.payload;
    if(reason!=='clickaway'){
        return;
    }else{
        yield put(applicationSetSnackbarStatus({open:false}));
    }
}

export function* handleToggleWindowState(){
    const windowState = yield select(getWindowState);
    if (windowState === 'maximized'){
        yield put(Window.actions.restore({}));
    }else if (windowState === 'normal'){
        yield put(Window.actions.maximize({}));
    }
}

export function* handleApplicationLaunchBarToggle(){
    const launchBarCollapse = yield select(getLaunchBarCollapse);

    yield put.resolve(Window.actions.getBounds({}));

    const getBoundsAction = yield take(Window.actions.GET_BOUNDS_RES);
    const getBoundsActionPayload = getBoundsAction.payload;

    if (window.location.href.toLowerCase().endsWith('launchbar')){
        // switch to main panel
        yield put.resolve(Window.actions.updateOptions({
            options:{
                resizable:true
            }
        }));
        yield put.resolve(Window.actions.setBounds({
            left:previousBaseWindow.left,
            top:previousBaseWindow.top,
            width:previousBaseWindow.width,
            height:previousBaseWindow.height,
        }));
        if (process.env.NODE_ENV !== 'test'){
            hist.push(previousBaseWindow.url);
        }
    }else{
        // switch to launchBar
        yield put.resolve(Window.actions.updateOptions({
            options:{
                resizable:false
            }
        }));
        previousBaseWindow.url = (new URL(window.location.href)).pathname;
        previousBaseWindow.top = getBoundsActionPayload.top;
        previousBaseWindow.left = getBoundsActionPayload.left;
        previousBaseWindow.width = getBoundsActionPayload.width;
        previousBaseWindow.height = getBoundsActionPayload.height;

        if (launchBarCollapse){
            yield put.resolve(Window.actions.setBounds({
                left:getBoundsActionPayload.left,
                top:getBoundsActionPayload.top,
                width:88,
                height:64,
            }));
        }else{
            yield put.resolve(Window.actions.setBounds({
                left:getBoundsActionPayload.left,
                top:getBoundsActionPayload.top,
                width:launchBarItems.length<10? launchBarItems.length*64+88:664,
                height:64,
            }));
        }
        if (process.env.NODE_ENV !== 'test'){
            hist.push('/launchBar');
        }

    }
}

export function* handleApplicationLaunchBarToggleCollapse() {
    const launchBarCollapse = yield select(getLaunchBarCollapse);

    yield put.resolve(Window.actions.getBounds({}));
    const getBoundsAction = yield take(Window.actions.GET_BOUNDS_RES);
    const getBoundsActionPayload = getBoundsAction.payload;

    if (launchBarCollapse){
        yield put.resolve(Window.actions.setBounds({
            left:getBoundsActionPayload.left,
            top:getBoundsActionPayload.top,
            width:88,
            height:64,
        }));
    }else{
        yield put.resolve(Window.actions.setBounds({
            left:getBoundsActionPayload.left,
            top:getBoundsActionPayload.top,
            width:launchBarItems.length<10? launchBarItems.length*64+88:664,
            height:64,
        }));
    }

}

export function* handleApplicationLaunchNewWindow(action) {
    const appJson = action.payload;
    const defaultWidth = yield select(getNewWindowWidth);
    const defaultHeight = yield select(getNewWindowHeight);
    const defaultTop = yield select(getNewWindowTop);
    const defaultLeft = yield select(getNewWindowLeft);

    if(!appJson.defaultWidth){ appJson.defaultWidth = defaultWidth}
    if(!appJson.defaultHeight){ appJson.defaultHeight = defaultHeight}
    if(!appJson.defaultTop){ appJson.defaultTop = defaultTop}
    if(!appJson.defaultLeft){ appJson.defaultLeft = defaultLeft}

    yield put.resolve(Window.actions.newWindow(appJson));

    yield put(configUpdateNewWindowPosition());

}

export default function* (){
    yield takeLatest(APPLICATION_STARTED,handleApplicationLoading);
    yield takeLatest(Event.actionDicts.windowEventDictByName['close-requested'].type,handleApplicationExit);
    yield takeLatest(APPLICATION_TOGGLE_WINDOW_STATE,handleToggleWindowState);
    yield takeLatest(APPLICATION_NEW_SNACKBAR,handleApplicationAddNewSnackBar);
    yield takeLatest(APPLICATION_CLOSE_SNACKBAR,handleApplicationCloseSnackBar);
    yield takeLatest(APPLICATION_LAUNCH_BAR_TOGGLE,handleApplicationLaunchBarToggle);
    yield takeLatest(APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE,handleApplicationLaunchBarToggleCollapse);
    yield takeLatest(APPLICATION_LAUNCH_NEW_WINDOW,handleApplicationLaunchNewWindow);
}