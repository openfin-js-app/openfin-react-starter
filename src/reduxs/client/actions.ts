import { createAction, ActionFunctionAny, Action } from 'redux-actions';
import { IClientState } from './types';

export const CLIENT_SET_VALUE:string = 'CLIENT_SET_VALUE';
export const clientSetValue:ActionFunctionAny<Action<Partial<IClientState>>> = createAction(CLIENT_SET_VALUE,
    (option:Partial<IClientState>)=>(option)
);