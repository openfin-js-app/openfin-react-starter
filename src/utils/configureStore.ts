import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createOpenfinMiddleware } from '@albertli90/redux-openfin';
import { ChannelType } from '@albertli90/redux-openfin/init';

import rootReducer, {IRootState} from '../reduxs';
import rootSaga from '../reduxs/sagas';

declare const window:any;

export default (
        channelType:ChannelType,
        channelClientId:string,
        sharedActions:string[],
        parentState?:IRootState
)=>{

    const openfinMiddleware = createOpenfinMiddleware(window.fin,{
        channelType,channelClientId,sharedActions,
        channelRandomSuffix:process.env.NODE_ENV === 'development',
        autoDocking:process.env.REACT_APP_ENABLE_AUTO_DOCKING === 'true',
        dockingOptions:{
            range: 120,
        }
    });
    const sagaMiddleware = createSagaMiddleware();
    const devtools = window.devToolsExtension?window.devToolsExtension():(f:any):any => (f);

    const middleware = compose(
        applyMiddleware(
            sagaMiddleware,
            openfinMiddleware,
        ),
        devtools
    );

    const store = createStore(
        rootReducer(parentState),
        middleware,
    );

    sagaMiddleware.run(rootSaga);

    return store;

}