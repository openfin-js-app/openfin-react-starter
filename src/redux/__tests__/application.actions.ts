import configureStore from 'redux-mock-store';
import * as Actions from '../application/actions';

const mockStore = configureStore();
const initialState = {};

const store = mockStore(initialState);

describe('Application actions',()=>{

    beforeEach(()=>{
        store.clearActions();
    });

    it('applicationStarted created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationStarted());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationChildStarted created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationChildStarted());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationReady created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationReady());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationDrawerToggle created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationDrawerToggle());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationToogleWindowState created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationToogleWindowState());
        expect(store.getActions()).toMatchSnapshot();
    });

    // snackbar

    it('applicationNewSnackbar created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationNewSnackbar({
            message:'Snackbar message',
            variant:'primary',
        }));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationSetSnackbarStatus created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationSetSnackbarStatus({
            open:false,
        }));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationProcessSnackbarQueue created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationProcessSnackbarQueue());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationCloseSnackbar created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationCloseSnackbar({
            event:{type:'sample event obj'},
            reason:'reason',
        }));
        expect(store.getActions()).toMatchSnapshot();
    });

    // launch bar

    it('applicationLaunchBarToggle created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationLaunchBarToggle());
        expect(store.getActions()).toMatchSnapshot();
    });


    it('applicationLaunchBarToggleCollapse created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationLaunchBarToggleCollapse());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationLaunchNewWindow created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationLaunchNewWindow({
            type: 'mock up app json obj'
        }));
        expect(store.getActions()).toMatchSnapshot();
    });

});