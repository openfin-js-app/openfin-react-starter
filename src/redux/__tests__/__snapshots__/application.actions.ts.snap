// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application actions applicationChildStarted created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_CHILD_STARTED",
  },
]
`;

exports[`Application actions applicationCloseSnackbar created and dispatched correctly 1`] = `
Array [
  Object {
    "payload": Object {
      "event": Object {
        "type": "sample event obj",
      },
      "reason": "reason",
    },
    "type": "APPLICATION_CLOSE_SNACKBAR",
  },
]
`;

exports[`Application actions applicationDrawerToggle created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_DRAWER_TOGGLE",
  },
]
`;

exports[`Application actions applicationLaunchBarToggle created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_LAUNCH_BAR_TOGGLE",
  },
]
`;

exports[`Application actions applicationLaunchBarToggleCollapse created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE",
  },
]
`;

exports[`Application actions applicationLaunchNewWindow created and dispatched correctly 1`] = `
Array [
  Object {
    "payload": Object {
      "type": "mock up app json obj",
    },
    "type": "APPLICATION_LAUNCH_NEW_WINDOW",
  },
]
`;

exports[`Application actions applicationNewSnackbar created and dispatched correctly 1`] = `
Array [
  Object {
    "payload": Object {
      "message": "Snackbar message",
      "variant": "primary",
    },
    "type": "APPLICATION_NEW_SNACKBAR",
  },
]
`;

exports[`Application actions applicationProcessSnackbarQueue created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_PROCESS_SNACKBAR_QUEUE",
  },
]
`;

exports[`Application actions applicationReady created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_READY",
  },
]
`;

exports[`Application actions applicationSetSnackbarStatus created and dispatched correctly 1`] = `
Array [
  Object {
    "payload": Object {
      "open": false,
    },
    "type": "APPLICATION_SET_SNACKBAR_STATUS",
  },
]
`;

exports[`Application actions applicationStarted created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_STARTED",
  },
]
`;

exports[`Application actions applicationToogleWindowState created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_TOGGLE_WINDOW_STATE",
  },
]
`;
